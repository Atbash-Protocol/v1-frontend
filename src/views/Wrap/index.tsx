import { Box, Grid, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';

import { BCard } from 'components/BCard';
import { BMultiTabs } from 'components/BMultiTab/BMultiTab';
import MemoInlineMetric from 'components/Metrics/InlineMetric';
import MenuMetric from 'components/Metrics/MenuMetric';
import { theme } from 'constants/theme';
import { selectFormattedStakeBalance } from 'store/modules/account/account.selectors';
import { selectFormattedIndex } from 'store/modules/stake/stake.selectors';

import { WrapBalanceMetrics } from './components/BalanceMetrics';
import { UnWrapAction } from './components/UnWrapAction';
import { WrapAction } from './components/WrapAction';

function Wrap() {
    const { t } = useTranslation();

    const currentIndex = useSelector(selectFormattedIndex);
    const balances = useSelector(selectFormattedStakeBalance);

    const wrapMetrics = [
        { key: t('common:YourBalance'), value: balances.BASH },
        { key: t('wrap:YourStakedBalance'), value: balances.SBASH },
    ].map(({ key: metricKey, value }, i) => <MemoInlineMetric key={`metric-${i}`} {...{ metricKey, value }} />);

    const tabs = [
        { label: t('Wrap:Wrap'), component: <WrapAction /> },
        { label: t('Wrap:UnWrap'), component: <UnWrapAction /> },
    ];

    return (
        <>
            <BCard title={t('wrap:WrapTitle')} zoom={true}>
                <Typography sx={{ color: theme.palette.primary.dark }}>
                    <>{t('wrap:WrapYourBASH')} </>
                </Typography>

                <Grid container pt={theme.spacing(4)}>
                    <Grid xs={2} item>
                        <MenuMetric metricKey={t('common:CurrentIndex')} value={currentIndex} />
                    </Grid>
                </Grid>

                <Box>
                    <BMultiTabs tabs={tabs} />
                </Box>

                <Box mt={theme.spacing(4)}>{wrapMetrics}</Box>
            </BCard>

            <BCard title={''} zoom={true}>
                <WrapBalanceMetrics />
            </BCard>
        </>
    );
}

export default Wrap;
