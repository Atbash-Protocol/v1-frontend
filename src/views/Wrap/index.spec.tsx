import { render } from '@testing-library/react';

import Wrap from '.';

describe('Wrap', () => {
    it('renders', () => {
        const comp = render(<Wrap />);

        expect(comp).toMatchSnapshot();
    });
});
