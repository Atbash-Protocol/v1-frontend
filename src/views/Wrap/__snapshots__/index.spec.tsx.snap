// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrap renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1vv71kp"
      >
        <div
          class="MuiBox-root css-19na2ny"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
          >
            wrap:WrapTitle
          </h4>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-dwvq3t-MuiTypography-root"
        >
          wrap:WrapYourBASH
           
        </p>
        <div
          class="MuiGrid-root MuiGrid-container css-12tl881-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-dvxtzn"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-y7tvl6-MuiTypography-root"
              >
                common:CurrentIndex
              </p>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ulaqe5-MuiTypography-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
                />
              </h5>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-1gsv261"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="basic tabs example"
                  class="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="bmultitab-tabpanel-0"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-k7y545-MuiButtonBase-root-MuiTab-root"
                    id="bmultitab-tab-0"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    Wrap:Wrap
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="bmultitab-tabpanel-0"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-k7y545-MuiButtonBase-root-MuiTab-root"
                    id="bmultitab-tab-0"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Wrap:UnWrap
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
            <div
              aria-labelledby="bmultitab-tab-0"
              id="bmultitab-tabpanel-0"
              role="tabpanel"
            >
              <div
                class="MuiBox-root css-1x5doh2"
              >
                WrapWrapAction
              </div>
            </div>
            <div
              aria-labelledby="bmultitab-tab-1"
              hidden=""
              id="bmultitab-tabpanel-1"
              role="tabpanel"
            >
              <div
                class="MuiBox-root css-1x5doh2"
              >
                UnWrapAction
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-h5fkc8"
        >
          <div
            class="MuiBox-root css-1z137pa"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
            >
              common:YourBalance
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
            >
              20.00
            </p>
          </div>
          <div
            class="MuiBox-root css-1z137pa"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
            >
              wrap:YourStakedBalance
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
            >
              20.00
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1vv71kp"
      >
        <div
          class="MuiBox-root css-19na2ny"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
          />
        </div>
        WrapBalanceMetrics
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1vv71kp"
    >
      <div
        class="MuiBox-root css-19na2ny"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
        >
          wrap:WrapTitle
        </h4>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-dwvq3t-MuiTypography-root"
      >
        wrap:WrapYourBASH
         
      </p>
      <div
        class="MuiGrid-root MuiGrid-container css-12tl881-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-dvxtzn"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-y7tvl6-MuiTypography-root"
            >
              common:CurrentIndex
            </p>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ulaqe5-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
              />
            </h5>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1gsv261"
        >
          <div
            class="MuiTabs-root css-1ujnqem-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                aria-label="basic tabs example"
                class="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="bmultitab-tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-k7y545-MuiButtonBase-root-MuiTab-root"
                  id="bmultitab-tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  Wrap:Wrap
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="bmultitab-tabpanel-0"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-k7y545-MuiButtonBase-root-MuiTab-root"
                  id="bmultitab-tab-0"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Wrap:UnWrap
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            aria-labelledby="bmultitab-tab-0"
            id="bmultitab-tabpanel-0"
            role="tabpanel"
          >
            <div
              class="MuiBox-root css-1x5doh2"
            >
              WrapWrapAction
            </div>
          </div>
          <div
            aria-labelledby="bmultitab-tab-1"
            hidden=""
            id="bmultitab-tabpanel-1"
            role="tabpanel"
          >
            <div
              class="MuiBox-root css-1x5doh2"
            >
              UnWrapAction
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            common:YourBalance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            20.00
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            wrap:YourStakedBalance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            20.00
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1vv71kp"
    >
      <div
        class="MuiBox-root css-19na2ny"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
        />
      </div>
      WrapBalanceMetrics
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
